src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs

```diff
@@ -108,139 +164,124 @@ public override void ConfigureServices(IServiceCollection services)
         public override void ConfigureServices(IServiceCollection services)
         {
             services.TryAddSingleton<IOpenIdValidationService, OpenIdValidationService>();
+            services.TryAddTransient<JwtBearerHandler>();
             services.AddScoped<IDisplayDriver<ISite>, OpenIdValidationSettingsDisplayDriver>();
 
-            services.TryAddScoped<IOpenIddictValidationEventService, OpenIddictValidationEventService>();
-            services.TryAddScoped<OpenIddictValidationHandler>();
-            services.TryAddScoped<OpenIddictValidationProvider>();
+            services.AddOpenIddict()
+                .AddValidation();
+
+            // Note: the OpenIddict extensions add an authentication options initializer that takes care of
+            // registering the validation handler. Yet, it MUST NOT be registered at this stage as it is
+            // lazily registered by OpenIdValidationConfiguration only after checking the OpenID validation
+            // settings are valid and can be safely used in this tenant without causing exceptions.
+            // To prevent that, the initializer is manually removed from the services collection of the tenant.
+            services.RemoveAll<IConfigureOptions<AuthenticationOptions>, OpenIddictValidationConfiguration>();
 
             // Register the options initializers required by OpenIddict and the JWT handler.
             services.TryAddEnumerable(new[]
             {
                 // Orchard-specific initializers:
-                ServiceDescriptor.Transient<IConfigureOptions<AuthenticationOptions>, OpenIdValidationConfiguration>(),
-                ServiceDescriptor.Transient<IConfigureOptions<JwtBearerOptions>, OpenIdValidationConfiguration>(),
-                ServiceDescriptor.Transient<IConfigureOptions<OpenIddictValidationOptions>, OpenIdValidationConfiguration>(),
+                ServiceDescriptor.Singleton<IConfigureOptions<AuthenticationOptions>, OpenIdValidationConfiguration>(),
+                ServiceDescriptor.Singleton<IConfigureOptions<JwtBearerOptions>, OpenIdValidationConfiguration>(),
+                ServiceDescriptor.Singleton<IConfigureOptions<OpenIddictValidationOptions>, OpenIdValidationConfiguration>(),
 
-                // Built-in initializers:
-                ServiceDescriptor.Transient<IPostConfigureOptions<JwtBearerOptions>, JwtBearerPostConfigureOptions>(),
-                ServiceDescriptor.Transient<IPostConfigureOptions<OpenIddictValidationOptions>, OpenIddictValidationInitializer>(),
-                ServiceDescriptor.Transient<IPostConfigureOptions<OpenIddictValidationOptions>, OAuthValidationInitializer>()
+                // Built-in initializers (note: the OpenIddict initializers are registered by AddValidation()).
+                ServiceDescriptor.Singleton<IPostConfigureOptions<JwtBearerOptions>, JwtBearerPostConfigureOptions>()
             });
         }
     }
+
+    internal static class OpenIdServiceCollectionExtensions
```

Comment:
I'm fairly confident these extensions could be useful in other modules.
@sebastienros let me know if you can think of a cool place for them.